script.py

```diff
@@ -0,0 +1,52 @@
+import os
+import re
+import pysrt
+import logging
+from pathlib import Path
+from pytube import YouTube
+
+logger = logging.getLogger(__name__)
+
+
+class Captions:
+    def __init__(self, URL):
+        self.yt = YouTube(URL)
+        self.video_path = os.getcwd() + "/videos/{}".format(
+            self.yt.title.lower().replace(" ", "_")
+        )
+        Path(self.video_path).mkdir(parents=True, exist_ok=True)
+        self.srt_filename = self.create_srt()
+        self.generate_script()
+
+    def create_srt(self):
+        try:
+            caption_dict = self.yt.captions.lang_code_index
```

Comment:
Couldn't you finish your `try ... except` block after this line? I'm not 100% sure but I think it's here where the captions "download" is made (or checked). This avoids the rest of the code to be indented